src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs

```diff
@@ -75,36 +76,43 @@ public class VsCredentialProviderImporter
             _initializer = initializer ?? Initialize;
         }
 
-        [Import("VisualStudioAccountProvider", typeof(IVsCredentialProvider), AllowDefault = true)]
-        public IVsCredentialProvider ImportedProvider { get; set; }
+        [ImportMany(typeof(IVsCredentialProvider))]
+        public IEnumerable<IVsCredentialProvider> ImportedProviders { get; set; }
 
         /// <summary>
         /// Plugin providers are entered loaded the same way as other nuget extensions,
         /// matching any extension named CredentialProvider.*.exe.
         /// </summary>
         /// <returns>An enumeration of plugin providers</returns>
-        public ICredentialProvider GetProvider()
+        public IReadOnlyCollection<ICredentialProvider> GetProviders()
         {
-            ICredentialProvider result = null;
+            var results = new List<ICredentialProvider>();
 
             try
             {
                 _initializer();
 
-                if (ImportedProvider != null)
+                if (ImportedProviders != null)
                 {
-                    result = new VsCredentialProviderAdapter(ImportedProvider);
+                    // Ensure imported providers ordering is deterministic
+                    foreach (var importedProvider in ImportedProviders.OrderBy(p => p.GetType().Name))
+                    {
+                        results.Add(new VsCredentialProviderAdapter(importedProvider));
+                    }
                 }
 
                 // Dev15+ will provide a credential provider for VSTS.
                 // If we are in Dev14, and no imported VSTS provider is found,
                 // then fallback on the built-in VisualStudioAccountProvider
-                if (result == null && IsDev14)
+                if (IsDev14
+                    && results.FirstOrDefault(p => p.Id.EndsWith(
```

Comment:
Changed in https://github.com/NuGet/NuGet.Client/pull/664/commits/a5a979f7192c3a68e4ca2f0f52ea07fa4d4f9435